### PR TITLE
fix atxi pagination OoB possibility

### LINKED ISSUE #626: atxi pagination param can cause slice OOB panic
```
panic: runtime error: slice bounds out of range

goroutine 8523 [running]:
github.com/ethereumproject/go-ethereum/core.GetAddrTxs.func1(0xc420164240, 0x8, 0x8, 0xc4710d7db0, 0x42, 0xc420164240)
	/Users/ia/gocode/src/github.com/ethereumproject/go-ethereum/core/atxi.go:374 +0x218
github.com/ethereumproject/go-ethereum/core.GetAddrTxs(0x4eb54c0, 0xc425687890, 0x4755566335d4bcf6, 0x4fc72826c4943649, 0xc442baaa33, 0x43d845, 0x46d182, 0x4cdd009, 0x1, 0x4cdd009, ...)
	/Users/ia/gocode/src/github.com/ethereumproject/go-ethereum/core/atxi.go:377 +0x54a
github.com/ethereumproject/go-ethereum/eth.(*PublicGethAPI).GetAddressTransactions(0xc4200bcab8, 0x4755566335d4bcf6, 0x4fc72826c4943649, 0x42baaa33, 0x43d845, 0x46d182, 0xc4367d9fea, 0x2, 0xc4367d9fec, 0x2, ...)
	/Users/ia/gocode/src/github.com/ethereumproject/go-ethereum/eth/api.go:1734 +0x374
reflect.Value.call(0xc420164000, 0xc4200bdaf8, 0x13, 0x4cddc5f, 0x4, 0xc4b71b0b60, 0x9, 0x9, 0x756ea19ac3b7f1, 0xc4b71b0b60, ...)
	/usr/local/Cellar/go/1.10.2/libexec/src/reflect/value.go:447 +0x969
reflect.Value.Call(0xc420164000, 0xc4200bdaf8, 0x13, 0xc4b71b0b60, 0x9, 0x9, 0x8, 0x8, 0x30)
	/usr/local/Cellar/go/1.10.2/libexec/src/reflect/value.go:308 +0xa4
github.com/ethereumproject/go-ethereum/rpc.(*Server).handle(0xc42afc93e0, 0x4eb17e0, 0xc4a6e7a420, 0x4eb7560, 0xc4fda1f3b0, 0xc42f2e0a20, 0xc4779fa200, 0x18, 0xc425ad0738)
	/Users/ia/gocode/src/github.com/ethereumproject/go-ethereum/rpc/server.go:330 +0x68c
github.com/ethereumproject/go-ethereum/rpc.(*Server).exec(0xc42afc93e0, 0x4eb17e0, 0xc4a6e7a420, 0x4eb7560, 0xc4fda1f3b0, 0xc42f2e0a20)
	/Users/ia/gocode/src/github.com/ethereumproject/go-ethereum/rpc/server.go:352 +0x1b0
github.com/ethereumproject/go-ethereum/rpc.(*Server).serveRequest.func2(0xc4666f7540, 0xc42afc93e0, 0xc4f0cd1650, 0x4eb7560, 0xc4fda1f3b0, 0xc422293c60, 0x1, 0x1, 0x0)
	/Users/ia/gocode/src/github.com/ethereumproject/go-ethereum/rpc/server.go:223 +0xad
created by github.com/ethereumproject/go-ethereum/rpc.(*Server).serveRequest
	/Users/ia/gocode/src/github.com/ethereumproject/go-ethereum/rpc/server.go:218 +0x291
```

### PR DESCRIPTION
**Pull request by whilei**
problem: atxi pagination param can cause slice OoB
solution: ensure param within len lim

Fixes #626

### CODE REVIEW

File: core/atxi.go [hunk 1]
```diff
@@ -367,9 +367,13 @@ func GetAddrTxs(db ethdb.Database, address common.Address, blockStartN uint64, b
 		}
 		if paginationStart < 0 {
 			paginationStart = 0
+		} else if paginationStart > len(s) {
+			paginationStart = len(s)
 		}
 		if paginationEnd < 0 {
```
Comment (tzdybal) [inline core/atxi.go]: This can be simplified to:
```
if paginationEnd < 0 || paginationEnd > len(s) {
    paginationEnd = len(s)
}
```

**Comment by whilei** [comment]
@tzdybal Fixed, and refactored `core/GetAtxi` method to return an error, if any. Also added an additional validation that `! paginationStart > paginationEnd`.

**Comment by whilei** [comment]
Maybe actually fixed.

**Comment by whilei** [comment]
Just a heads up that this failing test is #630, fixed with #632, and is unrelated to these changes.

**Comment by tzdybal** [comment]
:lgtm:

---

Reviewed 2 of 3 files at r2, 3 of 3 files at r3.
Review status: all files reviewed, 1 unresolved discussion (waiting on @tzdybal and @r8d8)

---



*Comments from [Reviewable](https://reviewable.io/reviews/ethereumproject/go-ethereum/627#-:-LF2NTLy7hV4PgSHZo0H:bnfp4nl)*

**Pull request merged**